volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,506 @@
+        if netlink_sock.groups != 0:
+            groups_bitmap = netlink_sock.groups.dereference()
+            groups_str = f"groups:0x{groups_bitmap:08x}"
+            saddr_list.append(groups_str)
+
+        daddr_list = []
+        dst_portid = f"portid:{netlink_sock.dst_portid}"
+        daddr_list.append(dst_portid)
+        dst_group = f"group:0x{netlink_sock.dst_group:08x}"
+        daddr_list.append(dst_group)
+        module = netlink_sock.module
+        if module and netlink_sock.module.name:
+            module_name_str = utility.array_to_string(netlink_sock.module.name)
+            module_name = f"lkm:{module_name_str}"
+            daddr_list.append(module_name)
+
+        saddr_tag = ",".join(saddr_list)
+        daddr_tag = ",".join(daddr_list)
+        state = netlink_sock.get_state()
+
+        sock_stat = saddr_tag, daddr_tag, state
+        return netlink_sock, sock_stat
+
+    def _vsock_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_VSOCK socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            vsock_sock: The kernel `vsock_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        vsock_sock = sock.cast("vsock_sock")
+        saddr = vsock_sock.local_addr.svm_cid
+        sport = vsock_sock.local_addr.svm_port
+        daddr = vsock_sock.remote_addr.svm_cid
+        dport = vsock_sock.remote_addr.svm_port
+        state = vsock_sock.get_state()
+
+        saddr_tag = f"{saddr}:{sport}"
+        daddr_tag = f"{daddr}:{dport}"
+        sock_stat = saddr_tag, daddr_tag, state
+        return vsock_sock, sock_stat
+
+    def _packet_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_PACKET socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            packet_sock: The kernel's `packet_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        packet_sock = sock.cast("packet_sock")
+        ifindex = packet_sock.ifindex
+        dev_name = self._netdevices.get(ifindex, "") if ifindex > 0 else "ANY"
+
+        saddr_tag = f"{dev_name}"
+        daddr_tag = ""
+        state = packet_sock.get_state()
+        sock_stat = saddr_tag, daddr_tag, state
+        return packet_sock, sock_stat
+
+    def _xdp_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_XDP socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            xdp_sock: The kernel's `xdp_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        xdp_sock = sock.cast("xdp_sock")
+        device = xdp_sock.dev
+        if not device:
+            return
```

Comment:
As above please don't bail out of the code when it would be easy to invert the guard and indent so that it's clear what runs when.